VP_VKD3D_requirements.json

```diff
@@ -0,0 +1,759 @@
+                "VkPhysicalDeviceVulkan11Features": {
+                    "shaderDrawParameters": true
+                },
+                "VkPhysicalDeviceVulkan13Features": {
+
+                },
+                "VkPhysicalDeviceRobustness2FeaturesEXT": {
+                    "nullDescriptor": true
+                },
+                "VkPhysicalDeviceExtendedDynamicStateFeaturesEXT": {
+                    "extendedDynamicState": true
+                },
+                "VkPhysicalDeviceExtendedDynamicState2FeaturesEXT": {
+                    "extendedDynamicState2": true
+                }
+            }
+        },
+        "d3d12_level_11_0_descriptor_sets": {
+            "properties": {
+                "VkPhysicalDeviceVulkan12Properties": {
+                    "maxPerStageDescriptorUpdateAfterBindSampledImages": 1000000,
+                    "maxPerStageDescriptorUpdateAfterBindStorageImages": 1000000,
+                    "maxPerStageDescriptorUpdateAfterBindStorageBuffers": 1000000
+                }
+            }
+        },
+        "d3d12_level_11_0_descriptor_buffers": {
+            "extensions": {
+                "VK_EXT_descriptor_buffer": 1
+            },
+            "features": {
+                "VkPhysicalDeviceDescriptorBufferFeaturesEXT": {
+                    "descriptorBuffer": true,
+                    "descriptorBufferPushDescriptors": true
+                }
+            },
+            "properties": {
+                "VkPhysicalDeviceProperties": {
+                    "limits": {
+                        "maxPerStageDescriptorStorageBuffers": 64,
+                        "maxPerStageDescriptorStorageImages": 64,
+                        "minStorageBufferOffsetAlignment": 4
+                    }
+                }
+            }
+        },
+        "d3d12_level_11_0_mutable_descriptor_type_ext": {
+            "extensions": {
+                "VK_EXT_mutable_descriptor_type": 1
+            },
+            "features": {
+                "VkPhysicalDeviceMutableDescriptorTypeFeaturesEXT": {
+                    "mutableDescriptorType": true
+                }
+            }
+        },
+        "d3d12_level_11_0_mutable_descriptor_type_valve": {
+            "extensions": {
+                "VK_VALVE_mutable_descriptor_type": 1
+            },
+            "features": {
+                "VkPhysicalDeviceMutableDescriptorTypeFeaturesVALVE": {
+                    "mutableDescriptorType": true   
+                }
+            }
+        },
+        "d3d12_level_11_1": {
+            "extensions": {
+            },
+            "features": {
+                "VkPhysicalDeviceFeatures": {
+                    "logicOp": true,
+                    "vertexPipelineStoresAndAtomics": true
+                }
+            },
+            "properties": {
+                "VkPhysicalDeviceProperties": {
+                    "limits": {
+                        "maxPerStageDescriptorStorageBuffers": 64,
```

Comment:
This is somewhat redundant since we rely on update-after-bind.

Reply:
This comes from here:
```
static void d3d12_device_caps_init_feature_level(struct d3d12_device *device)
{
    const VkPhysicalDeviceFeatures *features = &device->device_info.features2.features;
    const struct vkd3d_vulkan_info *vk_info = &device->vk_info;
    struct d3d12_caps *caps = &device->d3d12_caps;

    caps->max_feature_level = D3D_FEATURE_LEVEL_11_0;

    if (caps->options.OutputMergerLogicOp && features->vertexPipelineStoresAndAtomics &&
            vk_info->device_limits.maxPerStageDescriptorStorageBuffers >= D3D12_UAV_SLOT_COUNT &&
            vk_info->device_limits.maxPerStageDescriptorStorageImages >= D3D12_UAV_SLOT_COUNT)
        caps->max_feature_level = D3D_FEATURE_LEVEL_11_1;
```

So I am wondering, if this code is correct but it's hard for me to really know.